runtime/testy/studentAPI.py

```diff
@@ -114,6 +134,19 @@ def is_running(self, fn):
 
     return fn in self._coroutines_running
 
+  def _check_params(self, uid, param):
+    device_type = uid >> 72
+    valid_params = self.deviceType_to_validParams[device_type]
+    if param not in valid_params:
+      raise StudentAPITypeError("Invalid param passed in, valid parameters for this device are: " + "".join(valid_params))
+
+  def _check_value(self, param, value):
+    valid_values = self.param_to_valid_values[param]
+    if valid_values[0] == "boolean":
```

Comment:
so in your dictionary, instead of storing "bool", store the class `bool`
the name of the class as a string: `bool.__name__`
this way, you can do a more robust check of `if isinstance(value, bool)`
http://stackoverflow.com/questions/152580/whats-the-canonical-way-to-check-for-type-in-python